frontend/src/metabase/visualizations/lib/sync-settings.ts

```diff
@@ -99,11 +99,7 @@ function syncGraphMetricSettings(
     return settings;
   }
 
-  const metricColumnNames = new Set(
-    cols
-      .filter(column => column.source === "aggregation")
-      .map(column => column.name),
-  );
+  const metricColumnNames = new Set(cols.map(column => column.name));
```

Comment:
I think this is incorrect and will auto-add irrelevant columns.

Reply:
I'm thinking about something like this:
```
// keep metricColumnNames unchanged
const allColumnNames = new Set(cols.map(column => column.name));

// change removedMetricColumnNames
const removedMetricColumnNames = new Set(
    [...prevMetricColumnNames].filter(name => !allColumnNames.has(name)),
  );
```

Reply:
@ranquild thanks! I'll investigate